src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/AbstractRequest.java

```diff
@@ -447,4 +447,63 @@ public ResourceLocation getDisplayIcon()
         }
         return Optional.empty();
     }
+
+    @Override
+    public boolean equals(final Object o)
+    {
+        if (this == o)
+        {
+            return true;
+        }
+        if (!(o instanceof AbstractRequest))
+        {
+            return false;
+        }
+
+        final AbstractRequest<?> that = (AbstractRequest<?>) o;
+
+        if (!getToken().equals(that.getToken()))
+        {
+            return false;
+        }
+        if (!requested.equals(that.requested))
+        {
+            return false;
+        }
+        if (!getChildren().equals(that.getChildren()))
+        {
+            return false;
+        }
+        if (!getRequester().equals(that.getRequester()))
+        {
+            return false;
+        }
+        if (getState() != that.getState())
+        {
+            return false;
+        }
+        if (getResult() != null ? !getResult().equals(that.getResult()) : that.getResult() != null)
+        {
+            return false;
+        }
+        if (getParent() != null ? !getParent().equals(that.getParent()) : that.getParent() != null)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS864)